docs/adr/006_cups.md

```diff
@@ -0,0 +1,296 @@
+                    CUPS Server->>Concentrator: Updated cupsUri and cupsCred
+                else equal cupsCredCrc
+                    CUPS Server->>Concentrator: Updated cupsUri
+                end
+            else equal cupsUri
+                alt different cupsCredCrc
+                    CUPS Server->>Facade Function: Retrieve CUPS Credential Blob
+                    Facade Function->>CUPS Server: CUPS Credential Blob
+                    CUPS Server->>Concentrator: Updated cupsCred
+                else equal cupsCredCrc
+                    CUPS Server->>CUPS Server: Checks tcUri and tcCredCrc as per above flow and properly return an update
+                end
+            end
+        end
+    end
+```
+
+### IoT Hub related changes
+
+Two changes are expected for IoT Hub.
+
+First one is related to the ability to associate an Azure storage account with IoT Hub.
+
+One of the possible 'sinks' of where the credential files are going to be stored is the IoT Hub linked Azure Blob Storage.
+
+For this option to work, you must [associate an Azure Storage account to IoT Hub](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-csharp-csharp-file-upload#associate-an-azure-storage-account-to-iot-hub). Any LoRaWan Starter Kit IoT Hub, should have this option enabled by default, therefore some changes are required in the way the template is deployed.
+
+Second change is related to the concentrator device twin.
+
+The following "desired" properties should be properly set:
+
+```json
+"cups": {
+    "cupsUri": "https://IP_or_DNS:PORT",
+    "tcUri": "wss://IP_or_DNS:PORT",
+    "cupsCredCrc": INT,
+    "tcCredCrc": INT,
+    "cupsCredentialUrl": "https://...",
+    "tcCredentialUrl": "https://..."
+},
+// Following field to be used for first version client certificate validation
+"clientThumbprint": "Client certificate thumbprint"
+```
+
+- **'cupsCredCrc'**: should be computed as CRC32 checksum calculated over the concatenated credentials files `cups.{trust,cert,key}`
+- **'tcCredCrc'**: should be computed as CRC32 checksum calculated over the concatenated credentials files `tc.{trust,cert,key}`
+- **'cupsCredentialUrl'**: should point to the blob/secret containing the concatenated credentials `cups.{trust,cert,key}`
+- **'tcCredentialUrl'**: should point to the blob/secret containing the concatenated credentials `tc.{trust,cert,key}`
+- **'cupsClientThumbprint'**: should include the thumbprint of the client certificate used for authenticating against the CUPS server
+
+### Facade Azure Function related changes
+
+The Azure Function should implement a new endpoint 'FetchConcentratorCredentials' which executes the following flow:
+
+```mermaid
+sequenceDiagram
+LoRaWanNetworkSrvModule->>Function: POST /fetchConcentratorCredentials
+Note right of LoRaWanNetworkSrvModule: Request includes 'stationEui' and 'credentialsType' (cups/tc/both)
+Function->>IoT Hub: Retrieve Concentrator device Twin
```

Comment:
An alternative would be to make the secret URL as part of the request as we have it locally